client/my-sites/stats/stats-purchase/stats-purchase-tier-upgrade-slider.tsx

```diff
@@ -29,16 +39,34 @@ function useTranslatedStrings() {
 	};
 }
 
-function TierUpgradeSlider( { className }: TierUpgradeSliderProps ) {
+function TierUpgradeSlider( { className, priceTiers, currencyCode }: TierUpgradeSliderProps ) {
+	const translate = useTranslate();
+	const infoReferenceElement = useRef( null );
 	const componentClassNames = classNames( 'stats-tier-upgrade-slider', className );
+	const EXTENSION_THRESHOLD = 2; // in millions
 
-	// Get the plan details.
-	const plans = useAvailableUpgradeTiers();
+	// Transform plan details to dusplay.
```

Comment:
Could we move this logic up to `stats-purchase/index.tsx` or `stats-purchase/stats-purchase-single-item.tsx`?

My thinking was we'd do the page-specific translation logic at a higher level and just pass in the already translated/messaged strings and values to the component. In that way, we can easily use this same component for the PWYW flow.

If we put this logic directly in `TierUpgradeSlider` we'd need to duplicate the component for the updated PWYW flow. If we abstract the logic a bit, we could call the slider with some already translated UI strings, some "steps", and a callback and use it for both flows.

So instead of:
`
<TierUpgradeSlider priceTiers={ priceTiers } currencyCode={ currencyCode } />
`

…something like:
`
<TierUpgradeSlider
  lhCallout="Monthly site views limit"
  rhCallout="You will pay"
  infoMessage="Price per month, billed yearly"
  onChange={ handleOnChange }
  steps={ paidUpgradeTiers }
/>
`

Then for PWYW we can do almost the same thing:
`
<TierUpgradeSlider
  lhCallout="Your monthly contribution"
  rhCallout="Thank you!"
  infoMessage="The average person pays $5 per month, billed yearly"
  onChange={ handleOnChange }
  steps={ pwywUpgradeTiers }
/>
`

We'd then need to define shape of the `steps` data but maybe just an array of { lhValue, rhValue, infoMessage ) values or something similar?

Reply:
If it's just translations then it makes sense to do so, but when I think about it, there are marks, emojis and probably more differences for the two views. I guess we could do two atm, and refactor when we see a lot of duplication.

What do you say? @grzegorz-cp

Reply:
That's a great argument, and it made me go through the structure one more time.
I'm open to modification ideas, but after looking at the designs again, I think the PWYW slider and Commercial slider only share CSS, but they are actually not that similar.

TieredUpgradeSlider is already an abstraction layer over the Slider component, and I believe that the tier manipulation belongs there because it takes incoming data and prepares it for the format needed by the slider. Other components don't need that structure. 

I'm afraid that if we start moving that logic to the higher components, those components will start to grow and be hard to maintain - there will be more logic coming from other children to match the pattern. On the other hand, if we move the logic up and also update the TieredUpgradeSlider to handle multiple cases, we end up inflating that component as well (e.g. marks vs emoji, showing/hiding slider for the extension tiers, etc.). In the end, we'll get something similar to the StatsModule components that are on the grid below the chart of the traffic page. One component handles all of the different widgets but each widget is slightly different.